Makefile

```diff
@@ -33,8 +33,7 @@ build: export SPDK_GIT_BRANCH != git -C spdk name-rev --name-only HEAD
 build: export SPDK_GIT_COMMIT != git rev-parse HEAD:spdk
 build: export BUILD_DATE != date -u +"%Y-%m-%d %H:%M:%S %Z"
 build: export NVMEOF_GIT_MODIFIED_FILES != git status -s | grep -e "^ *M" | sed 's/^ *M //' | xargs
-build: export CEPH_CLUSTER_CEPH_REPO_BASEURL != curl -s https://shaman.ceph.com/api/repos/ceph/$(CEPH_BRANCH)/$(CEPH_SHA)/centos/9/ | jq -r '.[0].url'
-
+build: export CEPH_CLUSTER_CEPH_REPO_BASEURL != curl -s https://shaman.ceph.com/api/repos/ceph/$(CEPH_BRANCH)/$(CEPH_SHA)/centos/9/ | jq -r '.[] | select(.status == "ready" and .sha1 == "$(CEPH_SHA)") | .url'
```

Comment:
Ah, I didn't realize that you were already querying shaman.  I don't think this `select` does anything useful as both the current query and this `select` query can land on an arm64 job for which we only build but don't produce packages.  You don't actually select a repo but repo dump all of them.

I'd simply change the link to
> https://shaman.ceph.com/api/repos/ceph/$(CEPH_BRANCH)/$(CEPH_SHA)/centos/9/x86_64

(i.e. add an additional filter for the architecture) to make shaman return a redirect to the repo.  If you want to capture it in a variable you would need something like `curl -Ls -w %{url_effective} -o /dev/null <URL>` instead of just `curl -s <URL>`.  Or you could get rid of the variable and embed this link directly where it's used.

Reply:
@idryomov 👍 amended to select x86_64 arch results, this issue was noted also by @gbregman